modules/artifact/src/main/java/org/mule/module/artifact/classloader/DefaultResourceReleaser.java

```diff
@@ -188,17 +195,60 @@ private boolean hasDeclaredMethod(Class<?> clazz, String methodName, Class<?>...
     try {
-      classAbandonedConnectionCleanupThread =
-          this.getClass().getClassLoader().loadClass("com.mysql.jdbc.AbandonedConnectionCleanupThread");
+      for (String knownCleanupThreadClassAddress : CONNECTION_CLEANUP_THREAD_KNOWN_CLASS_ADDRESES) {
+        try {
+          classAbandonedConnectionCleanupThread =
+              this.getClass().getClassLoader().loadClass(knownCleanupThreadClassAddress);
+        } catch (ClassNotFoundException e) {
+          logger.warn("No AbandonedConnectionCleanupThread registered with class address " + knownCleanupThreadClassAddress);
+          continue;
+        }
+      }
+      if (classAbandonedConnectionCleanupThread == null) {
+        logger.warn("No AbandonedConnectionCleanupThread class found. Check if thread class is between "
+            + CONNECTION_CLEANUP_THREAD_KNOWN_CLASS_ADDRESES.toString());
+        throw new ClassNotFoundException(CONNECTION_CLEANUP_THREAD_KNOWN_CLASS_ADDRESES.get(0));
+      }
       try {
         Method uncheckedShutdown = classAbandonedConnectionCleanupThread.getMethod("uncheckedShutdown");
         uncheckedShutdown.invoke(null);
       } catch (NoSuchMethodException e) {
         Method checkedShutdown = classAbandonedConnectionCleanupThread.getMethod("shutdown");
         checkedShutdown.invoke(null);
       }
+
+      try {
+        // In new mysql driver versions (at least 8), the executor service is wrapped inside a delegate class
+        // (DelegatedExecutorService) that
+        // exposes only the ExecutorService interface. Extract real ThreadPoolExecutor, which has the single threaded
+        // threadFactory saved, to clean it after disposal
+
+        // Hierarchy leading to real ThreadPoolExecutor
+        // AbandonedConnectionCleanupThread.cleanupThreadExecutorService -> class DelegatedExecutorService.e -> class
+        // ThreadPoolExecutor
+
+        // Note that the field 'cleanupThreadExcecutorService' is mispelled. There's actually a typo in MySql driver
+        // code.
+        Field cleanupExecutorServiceField = classAbandonedConnectionCleanupThread
+            .getDeclaredField("cleanupThreadExcecutorService");
+        cleanupExecutorServiceField.setAccessible(true);
+        ExecutorService delegateCleanupExecutorService =
+            (ExecutorService) cleanupExecutorServiceField.get(classAbandonedConnectionCleanupThread);
+
+        Field realExecutorServiceField = delegateCleanupExecutorService.getClass().getSuperclass().getDeclaredField("e");
+        realExecutorServiceField.setAccessible(true);
+        ThreadPoolExecutor realExecutorService =
+            (ThreadPoolExecutor) realExecutorServiceField.get(delegateCleanupExecutorService);
+
+        realExecutorService.setThreadFactory(Executors.defaultThreadFactory());
+      } catch (NoSuchFieldException e) {
+        logger.warn("Error cleaning threadFactory from AbandonedConnectionCleanupThread executor service");
+        e.printStackTrace();
```

Comment:
![Vulnerability](https://sonarsource.github.io/sonar-github/vulnerability.png 'Vulnerability') Vulnerability: Use a logger to log this exception. ([squid:S1148](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS1148))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7685&issues=AWjzLS16otIXnmsUYkAt&open=AWjzLS16otIXnmsUYkAt)

[ISSUE_KEY]: AWjzLS16otIXnmsUYkAt